src/main/java/org/jenkinsci/plugins/workflow/steps/CoreWrapperStep.java

```diff
@@ -148,8 +166,23 @@ private void doStart() throws Exception {
             this.disposer = disposer;
         }
 
-        @Override protected void finished(StepContext context) throws Exception {
-            disposer.tearDown(context.get(Run.class), context.get(FilePath.class), context.get(Launcher.class), context.get(TaskListener.class));
+        @Override protected void finished(@Nonnull StepContext context) throws Exception {
+            // Get the context objects that may be required
+            final Run<?,?> run = context.get(Run.class);
+            final FilePath workspace = context.get(FilePath.class);
+            final Launcher launcher = context.get(Launcher.class);
+            final TaskListener listener = context.get(TaskListener.class);
+            // Ensure those that are required are provided
+            Objects.requireNonNull(run);
+            if (this.disposer.requiresWorkspace() && workspace == null) {
+                throw new MissingContextVariableException(FilePath.class);
+            }
+            if (this.disposer.requiresLauncher() && launcher == null) {
+                throw new MissingContextVariableException(Launcher.class);
+            }
```

Comment:
As noted in https://github.com/jenkinsci/jenkins/pull/4784#discussion_r445877019 this whole `finished` method could just be reverted to the original.

Reply:
Updated to a recent incremental, so these are gone